src/edge/edgeManager.ts

```diff
@@ -177,6 +177,95 @@ export class EdgeManager {
         }
     }
 
+    public async loadWebView(outputChannel: vscode.OutputChannel) {
+      const panel = vscode.window.createWebviewPanel(Constants.galleryPanelViewType, Constants.galleryPanelViewTitle, vscode.ViewColumn.One, {
+        enableScripts: true,
+        retainContextWhenHidden: true,
+      });
+
+      const srcPath: string = this.context.asAbsolutePath(path.join(Constants.assetsFolder, Constants.galleryAssetsFolder, Constants.galleryIndexHtmlName));
+      panel.webview.html = await this.getWebViewContent(srcPath);
+
+      // Handle messages from the webview
+      panel.webview.onDidReceiveMessage(async (message) => {
+        switch (message.command) {
+          case "openSample":
+            if (message.name && message.url) {
+              await vscode.commands.executeCommand("azure-iot-edge.initializeSample", message.name, message.url, message.platform);
+            }
+            break;
+          case "openLink":
+            if (message.url) {
+              TelemetryClient.sendEvent(Constants.openSampleUrlEvent, {Url: message.url});
+              await vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(message.url));
+            }
+            break;
+        }
+      }, undefined, this.context.subscriptions);
+    }
+
+    public async getWebViewContent(templatePath: string): Promise<string> {
+      const dirPath = path.dirname(templatePath);
+      let html = await fse.readFile(templatePath, "utf-8");
+
+      html = html.replace(/(<link.*?\shref="|<script.*?\ssrc="|<img.*?\ssrc=")(.+?)"/g, (m, $1, $2) => {
+          return $1 + vscode.Uri.file(path.join(dirPath, $2)).with({ scheme: "vscode-resource" }).toString(true) + "\"";
+      });
+      return html;
+    }
+
+    public async initializeSample(name: string, url: string, platform: string, channel: vscode.OutputChannel) {
+        try {
+            const parentPath = await this.getSolutionParentFolder();
+
+            if (parentPath === undefined) {
+                throw new UserCancelledError();
+            }
+
+            await fse.ensureDir(parentPath);
+            const sampleName: string = await this.inputSolutionName(parentPath, name);
+            const samplePath: string = path.join(parentPath, sampleName);
+
+            channel.show();
+            channel.appendLine(`Downloading sample project to ${samplePath}...`);
+            await this.downloadSamplePackage(url, samplePath);
+
+            const defaultPlatformKey = Utility.getVscodeSettingKey(Constants.defPlatformConfig);
+            let vscodeSettingJson = await Utility.getUserSettingJsonFromSolutionPath(samplePath);
+
+            if (vscodeSettingJson === undefined) {
+                vscodeSettingJson = {};
+            }
+
+            if (vscodeSettingJson[defaultPlatformKey] === undefined) {

```

Comment:
how about !vscodeSettingJson[defaultPlatformKey] to also include the null case

Reply:
Thanks, good suggestion